compiler/testData/codegen/bytecodeText/whenStringOptimization/duplicatingItemsSameHashCode2.kt

```diff
@@ -0,0 +1,17 @@
+// IGNORE_BACKEND: JVM
+import kotlin.test.assertEquals
+
+fun foo(x : String) : String {
+    assert("abz]".hashCode() == "aby|".hashCode())
+
+    when (x) {
+        "abz]" -> return "abz"
+        "ghi"  -> return "ghi"
+        "aby|" -> return "aby"
+        "abz]" -> return "fail"
+    }
+
+    return "other"
+}
+
+// 0 LOOKUPSWITCH
```

Comment:
Is LOOKUPSWITCH is absent cause of hashAndSwitchLabels.size <= 2 condition?

Reply:
Yes. I've updated the comments in it.

Reply:
Please also add box test for such case

Reply:
Didn't add it because it is exactly the same to the existing test. Will update after the tests finished.